packages/date-picker/src/DatePicker/DatePickerMenu/DatePickerMenuHeader/DatePickerMenuHeader.tsx

```diff
@@ -109,6 +126,8 @@ export const DatePickerMenuHeader = forwardRef<
             updateMonth(newMonth);
           }}
           className={selectInputWidthStyles}
+          onEntered={() => setIsSelectOpen(true)}
```

Comment:
If we go with this, then we don't even need the popoverProvider anymore since we're updating the state with the callbacks